query-lab/main.js

```diff
@@ -0,0 +1,163 @@
+const rl = readline.createInterface({
+	input: process.stdin,
+	output: process.stdout
+});
+
+const url = `mongodb://localhost:27017/crunchbase`
+
+mongoClient.connect(url, (error, db) => {
+  if (error) {
+    console.log('Error trying to connect to the Database');
+    console.log(error);
+  } else {
+    console.log('Connection established correctly!! 😬');
+
+    function mainMenu(){
+        clear();
+        printMenu();
+        rl.question('Type an option: ', (option) => {
+          switch(option){
+            case "1":
+                db.collection('companies').find({}, {name: 1, _id: 0}).toArray((error, result) => {
+                    if (error) {
+                      console.log(error);
+                      rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                    } else {
+                      console.log(result);
+                      rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                    }
+                  })
+              break;
+            case "2":
+                db.collection('companies').find({}, {name: 1, _id: 0}).toArray((error, result) => {
+                    if (error) {
+                      console.log(error);
+                      rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                    } else {
+                      console.log(result.length);
+                      rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                    }
+                  })
+              break;
+            case "3":
+                db.collection('companies').find({ founded_year: 2004 }, {name: 1, _id: 0}).toArray((error, result) => {
+                    if (error) {
+                      console.log(error);
+                      rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                    } else {
+                      console.log(result.length);
+                      rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                    }
+                })
+              break;
+            case "4":
+              db.collection('companies').find({ $and: [{ founded_year: 2004 }, { founded_month: 2 }]} , {name: 1, _id: 0}).toArray((error, result) => {
+                if (error) {
+                  console.log(error);
+                  rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                } else {
+                  console.log(result.forEach(elem => console.log(elem.name)));
+                  rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                }
+            })
+              break;
+            case "5":
+              db.collection('companies').find({ $and: [{ founded_year: 2004 }, { $and:[{founded_month:{$gte: 4}},{founded_month:{$lte: 6}} ] } ] }, {name: 1, _id: 0}).sort({founded_month: 1, founded_day: 1}).toArray((error, result) => {
```

Comment:
Solo un comentario, pues para esta query  al estar compuesta con condiciones para más de un campo en los documentos de la colección. La  decision de utilizar la conjunción lógica AND  es la correcta, sin embargo podría emplearse simplemente de la siguiente manera: 
```
   .find(
                {
                  $and: [
                    { founded_month: { $gte: 4 } },
                    { founded_month: { $lte: 6 } },
                    { founded_year: 2004 },
                  ],
                },
```